src/content/en/updates/2018/09/asynchronous-access-to-http-cookies.md

```diff
@@ -0,0 +1,187 @@
+  each application gets its own sandbox, and is completely independent from
+  other applications that the user might be running.
+  [Cookie scopes](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#Scope_of_cookies)
+  make for a significantly more complex security story, and merely attempting to
+  summarize that would double the size of this article.
+
+* Cookies have high performance costs. Browsers need to include a snapshot of
+  your cookies in every HTTP request, so every change to cookies must be
+  propagated across the storage and network stacks. Modern browsers have highly
+  optimized cookie store implementations, but we'll never be able to make
+  cookies as efficient as the other storage mechanisms, which don't need to talk
+  to the network stack.
+
+For all the reasons above, modern Web applications should avoid cookies and
+instead store a session identifier into
+[IndexedDB](https://developer.mozilla.org/en-US/docs/Web/API/IndexedDB_API), and
+explicitly add the identifier to the header or body of specific HTTP requests,
+via the [fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) API.
+
+That being said, you're still reading this article because you have a good
+reason to use cookies...
+
+## Say goodbye to jank
+
+The venerable
+[document.cookie](https://developer.mozilla.org/en-US/docs/Web/API/Document/cookie)
+API is a fairly guaranteed source of jank for your application. For example,
+whenever you use the `document.cookie` getter, the browser has to stop executing
+JavaScript until it has the cookie information you requested. This can take a
+process hop or a disk read, and will cause your UI to jank.
+
+A straightforward fix for this problem is switching from the `document.cookie`
+getter to the asynchronous Cookie Store API.
+
+    await cookieStore.get('session_id')
+
+    // {
+    //   domain: "example.com",
+    //   expires: 1593745721000,
+    //   name: "sesion_id",
+    //   path: "/",
+    //   sameSite: "unrestricted",
+    //   secure: true,
+    //   value: "yxlgco2xtqb.ly25tv3tkb8"
+    // }
+
+The `document.cookie` setter can be replaced in a similar manner. Keep in mind
+that the change is only guaranteed to be applied after the Promise returned by
+`cookieStore.set` resolves.
+
+    await cookieStore.set({ name: 'opt_out', value: '1' });
+
+    // undefined
+
+## Observe, don't poll
+
+A popular application for accessing cokies from JavaScript is detecting when
+the user logs out, and updating the UI. This is currently done by polling
+`document.cookie`, which introduces jank and has a negative impact on battery
+life.
+
+The Cookie Store API brings an alternative method for observing cookie
+changes, which does not require polling.
+
+    cookieStore.addEventListener('change', (event) => {
+      for (const cookie in event.changed) {
+        if (cookie.name === 'session_id')
+          SessionCookieChanged(cookie.value);
```

Comment:
I believe convention here is to go with lowercase leading characters for function names.

Reply:
Done